### PR TITLE
Static CMS image urls when using a different (sub)domain for custom admin url

### LINKED ISSUE #32930: Static CMS image urls when using a different (sub)domain for custom admin url
### Preconditions (*)

1. Magento 2.4.2

### Steps to reproduce (*)

1. Admin panel » Stores » Configuration » General » Content Management » Use Static URLs for Media Content in WYSIWYG: No
2. Admin panel » Stores » Configuration » Advanced » Admin » Admin Base URL » Use Custom Admin URL: Yes
3. Set admin to a different (sub)domain than the actual store url.
4. Add/edit a CMS page, static block or product page
5. Add image

### Expected result (*)

1. Media tag with relative image url
2. ``

### Actual result (*)

1. Media tag with static Magento admin url instead of store url
2. ``

This issue is caused by inconsistend use of `$this->_storeManager->getStore()` in /vendor/magento/module-cms/Helper/Wysiwyg/Images.php
Line 214: `$this->_storeManager->getStore()`
Line 299: `$this->_storeManager->getStore($this->_storeId)`

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by engcom-Kilo**
### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/32930

### Manual testing scenarios (*)

1. see https://github.com/magento/magento2/issues/32930

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@engcom-Kilo Please look at failed integration test

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
> @engcom-Kilo Please look at failed integration test

@Den4ik Fixed

**Review by Den4ik — APPROVED** [review]
✅ Approved.

@engcom-Kilo Thanks for the update

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, thank you for the review.
[ENGCOM-9107](https://jira.corp.magento.com/browse/ENGCOM-9107) has been created to process this Pull Request
:eight_spoked_asterisk: @Den4ik, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

File: app/code/Magento/Cms/Helper/Wysiwyg/Images.php [hunk 1]
```diff
@@ -41,51 +57,51 @@ class Images extends \Magento\Framework\App\Helper\AbstractHelper
     protected $_storeId;
 
     /**
-     * @var \Magento\Framework\Filesystem\Directory\Write
+     * @var Write
```
Comment (orlangur) [inline app/code/Magento/Cms/Helper/Wysiwyg/Images.php:60]: @engcom-Kilo please avoid unrelated changes.

**Comment by engcom-Alfa** [comment]
Hello @orlangur and @Den4ik,
Thank you for your contribution and collaboration!

I would like to test this Pull Request to unblock and speed up the delivery. 
Unfortunately, I am getting my instance ruined by while reproducing by looking into the [Issue Description](https://github.com/magento/magento2/issues/32930#issue-874404425) and the [Magento Admin Base URL Doc](https://docs.magento.com/user-guide/configuration/advanced/admin.html#admin-base-url).
Could you please provide more details on how it can be tested?

Below are the steps I am following to reproduce
- Set up the new instance using the admin base URL : local.magento2pr.com
- Admin side: Stores » Configuration » General » Content Management » `Use Static URLs for Media Content in WYSIWYG: No` (This is the default setting)
- Navigated to Admin panel » Stores » Configuration » Advanced » Admin » Admin Base URL and set the below settings shown in screenshot 
image

Here, After setting to new domain and admin path, the instance logs out as expected. Updated the system host and virtual host with the new domain. Still unable to regain the instance!

Kindly update me the complete steps to reproduce the issue, elaborating the step number 3 in the [Issue Description](https://github.com/magento/magento2/issues/32930#issue-874404425).

Thank you in advance!

**Comment by engcom-Alfa** [comment]
Hello @orlangur and @Den4ik,

Tried reproducing the issue in `2.4-develop` version before testing the given PR fix code referring PR description/ comments.
Issue is not reproducible and below is the information on this:


**Pre Conditions**

1. Installed a fresh 2.4-develop version in my local with sample data.


**Reproducible steps**

- Set up the new instance using the admin base URL as _local.magento2.com_
- Created a new category in the catalog and uploaded one image to it in the page content section.
- Admin side: Stores » Configuration » General » Content Management » `Use Static URLs for Media Content in WYSIWYG: No` (This is the default setting)
- Navigated to Admin panel » Stores » Configuration » Advanced » Admin » Admin Base URL and set the below settings shown in screenshot providing new sub domain as _admin.magento2.com_
> image
- Updated the system host and virtual host with the new admin domain.
- After restarting the system, Validated the image source domain in admin > Catalog > category page.
- Edited to new image and checked the image source domain.


**Expected Issue as per PR description/ comments:** 
> New edited sub domain is not reflected in the image source, old domain is captured

**Actual Result**
> Getting a proper sub domain image source is displayed.
> Below are the actual screenshot
> image
> image


**Kindly let me know if there is anything else involved to reproduce apart from above procedure or kindly cross check if it is particular to any specific case!**

**Comment by Den4ik** [comment]
Hi @engcom-Alfa 
I remember that I reproduced it but don't remember all steps. I'll try to reproduce it today or tomorrow and back with instruction

**Comment by engcom-Alfa** [comment]
Hi @Den4ik ,
Is it possible to provide me the steps to reproduce please?

**Comment by Den4ik** [comment]
@engcom-Alfa Sorry for delay. I hope to do this in the near future

**Pull request merged**